src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultHtmlFactsService.cs

```diff
@@ -19,15 +19,15 @@ public override bool TryGetElementInfo(SyntaxNode element, out SyntaxToken conta
                     return true;
                 case MarkupEndTagSyntax { Parent: MarkupElementSyntax parent } endTag:
                     containingTagNameToken = endTag.Name;
-                    attributeNodes = parent.StartTag.Attributes;
+                    attributeNodes = parent.StartTag?.Attributes ?? new SyntaxList<RazorSyntaxNode>();
```

Comment:
This is the specific fix. Everything else in this PR is noise / reaction to not expecting StartTag to be non-null.